chunk/chunk_store.go

```diff
@@ -565,7 +551,19 @@ func (c *AWSStore) lookupChunks(ctx context.Context, userID string, from, throug
 			if err != nil {
 				incomingErrors <- err
 			} else {
-				incomingChunkSets <- incoming
+				// Filter out chunks that are not in the selected time range.
+				filtered := make([]Chunk, 0, len(incoming))
+				for _, chunk := range incoming {
+					_, chunkFrom, chunkThrough, err := parseChunkID(chunk.ID)
+					if err != nil {
+						incomingErrors <- err
+					}
+					if chunkThrough < from || through < chunkFrom {
+						continue
+					}
+					filtered = append(filtered, chunk)
+				}
+				incomingChunkSets <- filtered
```

Comment:
I like the move, but can we move it further down in this function?  I like to keep the logic for parallelisation (the channels etc inside this look) nice and separate.